treewide: deprecate ip-up.target

groxxda opened the pull request:
###### Motivation for this change

Systemd upstream provides targets for networking. This also includes a target `network-online.target`.

I would like to see `ip-up.target` completely gone.

In this PR I remove / replace most occurrences since some of them were even wrong and could delay startup. 

There are still references that I'm unsure about:
- `nixos/modules/services/networking/dhcpcd.nix`: is this hook really needed?
- `nixos/modules/services/networking/avahi-daemon.nix`: has ordering `wantedBy` and `before` `if-up.target` which looks plain wrong to me. We should get `avahi-daemon` closer to upstream definition that also includes socket-activation, but this is out of scope of this PR
- `nixos/modules/services/networking/networkmanager.nix`: is this really needed?
- `nixos/modules/tasks/network-interfaces-scripted.nix`: is this needed?
- `nixos/modules/virtualisation/nova.nix`: this module still uses the `jobs` interface, which was deprecated when I started using nixos
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I would be thankful for comments about the remaining occurrences as well as deprecating `ip-up.target`  in general.

groxxda committed:
Add comment that ip-up target is deprecated

groxxda committed:
Replace most usages of ip-up.target in modules

groxxda committed:
networkd: remove ip-up.target

grahamc commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 14 of 14 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[nixos/modules/services/network-filesystems/drbd.nix, line 58 \[r1\]](https://reviewable.io:443/reviews/nixos/nixpkgs/18319#-KQuaZu4Bt6tHsnc2WG5:-KQuaZu4Bt6tHsnc2WG6:b-bhtydy) ([raw file](https://github.com/nixos/nixpkgs/blob/c03fbd54a2f894a8d466338a3ea015e20256701e/nixos/modules/services/network-filesystems/drbd.nix#L58)):*
> ```Nix
>       after = [ "systemd-udev.settle.service" "network.target" ];
>       wants = [ "systemd-udev.settle.service" ];
>       wantedBy = [ "multi-user.target" ];
> ```

Verified upstream: https://apt-browse.org/browse/ubuntu/trusty/main/i386/drbd8-utils/2:8.4.4-1ubuntu1/file/lib/systemd/system/drbd.service

---

*[nixos/modules/virtualisation/azure-agent.nix, line 182 \[r1\]](https://reviewable.io:443/reviews/nixos/nixpkgs/18319#-KQuaKmUyvNkkEPx_0Ss:-KQuaKmUyvNkkEPx_0St:batzxon) ([raw file](https://github.com/nixos/nixpkgs/blob/c03fbd54a2f894a8d466338a3ea015e20256701e/nixos/modules/virtualisation/azure-agent.nix#L182)):*
> ```Nix
>       wantedBy = [ "multi-user.target" ];
>       after = [ "network-online.target" "sshd.service" ];
>       wants = [ "network-online.target" ];
> ```

I was suspicious this needed to be up sooner, but verified otherwise: https://github.com/Azure/WALinuxAgent/blob/master/init/coreos/cloud-config.yml

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/18319#-:-KQub55jcepjWJ_hOIOa:bnfp4nl)*
<!-- Sent from Reviewable.io -->

edolstra commented:
Looks good to me.

What did you mean by the dhcpcd hook? The `start ip-up.target` can obviously be removed, but the `try-restart` of ntpd is still necessary unless ntpd has suddenly learned how to detect interface changes.

groxxda committed:
dhcpcd: remove ip-up

groxxda committed:
networkmanager: remove ip-up.target

groxxda committed:
network-interfaces-scripted: replace ip-up with network-online

groxxda commented:
Rebased to resolve merge conflicts and also added modified `dhcpcd` `networkmanager` and `network-interfaces-scripted`.
Please review whether the new commits look semantically good to you, I'm not an expert for the scripted networking stuff.

grahamc commented:
Reviewed 3 of 3 files at r3.
Review status: 16 of 17 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/18319)*
<!-- Sent from Reviewable.io -->

grahamc commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> but untested

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/18319#-:-KR1OCTeuPeK0UzpdPeo:bwjo6yg)*
<!-- Sent from Reviewable.io -->

joachifm commented:
LGTM. Anything holding this back?

joachifm commented:
Just a heads up: some networking related tests are now failing https://hydra.nixos.org/eval/1292016#tabs-now-fail

fpletz commented:
I'm investigating.

groxxda commented:
Thanks @fpletz. Feel free to revert until I find time to look into this.
By the way, is there a sane way to attach to a test vm to debug the failure?

fpletz commented:
@groxxda Found some info on https://nixos.org/nixos/manual/index.html#sec-running-nixos-tests and was able to debug with the execute function.

Should be fixed with https://github.com/NixOS/nixpkgs/commit/c58654e2b78e7d3ae97ba96c80ceb3f96236e463.

groxxda commented:
Thank you @fpletz for looking into this and fixing it!
I'll bring some :beers: when we meet next time.
